### DISCUSSION TITLE
Deduplication of class names from third party lib

### DISCUSSION THREAD

**Discussion by tcolinpa** [tailwindlabs/tailwindcss#11390] [Help]
I have a lib that I built on top of shadcn/ui and in the dist folder there's a index.css that contains all classes from tailwind that this lib uses.

When I use this lib in a project, I set up the globals.css with something like
```css
@tailwind base;
@tailwind components;
@tailwind utilities;

@import "@quaddro/react-ui/dist/index.css"; // this lib 
``` 

However, if I use the any tailwind utility classes such as 'relative', this gets duplicated in my final css bundle of this project because there's a 'relative' class inside @quaddro/react-ui/dist/index.css and then another one in the project's css (different lines on layout.css)
![image](https://github.com/tailwindlabs/tailwindcss/assets/44402590/e3ca0187-9388-46fe-8404-9bd643776125)

Is this expected behavior or did I mess it up?

Thanks!

**Comment by wongjn** [comment]
This would be behavior I'd expect.

**Reply by tcolinpa** [reply]
Is there a workaround this?

**Reply by wongjn** [reply]
You could consider leaving Tailwind CSS compilation to the consumer project.
